Runtimes/Core/SwiftOnoneSupport/CMakeLists.txt

```diff
@@ -0,0 +1,36 @@
+add_library(swiftSwiftOnoneSupport
+  SwiftOnoneSupport.swift
+  "${PROJECT_SOURCE_DIR}/linker-support/magic-symbols-for-install-name.c")
+
+set_target_properties(swiftSwiftOnoneSupport PROPERTIES Swift_MODULE_NAME SwiftOnoneSupport)
+
+if(APPLE AND BUILD_SHARED_LIBS)
+  target_link_options(swiftSwiftOnoneSupport PRIVATE "SHELL:-Xlinker -headerpad_max_install_names")
+endif()
+
+target_compile_options(swiftSwiftOnoneSupport PRIVATE
+  $<$<COMPILE_LANGUAGE:Swift>:-parse-stdlib>
+  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-Xllvm -sil-inline-generics=false>"
+  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-Xfrontend -validate-tbd-against-ir=none>"
+  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-Xfrontend -check-onone-completeness>"
+  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-Xfrontend -disable-access-control>"
+)
+
+target_compile_definitions(swiftSwiftOnoneSupport PRIVATE
+  $<$<COMPILE_LANGUAGE:C,CXX>:-DSWIFT_TARGET_LIBRARY_NAME=swiftSwiftOnoneSupport>
+)
+
+target_link_libraries(swiftSwiftOnoneSupport
+  PRIVATE
+    swiftShims
+    swiftCore)
+
+install(TARGETS swiftSwiftOnoneSupport)
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/SwiftOnoneSupport.swiftmodule"
+  DESTINATION "${CMAKE_INSTALL_LIBDIR}/swift/SwiftOnoneSupport.swiftmodule"
+  RENAME "${SwiftCore_MODULE_TRIPLE}.swiftmodule")
```

Comment:
This needs to be tweaked for the correct path as I've done in #78722.

Reply:
That seems pretty close. I'm waiting on that to patch up the change for Concurrency as well.

Reply:
https://github.com/swiftlang/swift/pull/78860 landed. We can use `${SwiftCore_INSTALL_SWIFTMODULEDIR}/SwiftOnoneSupport.swiftmodule` for the `DESTINATION` directory.

Reply:
This is tackled with [bca4796](https://github.com/swiftlang/swift/commit/bca4796d25d63eb82cbbdf27325790d839b53087)